### DISCUSSION TITLE
Literal tokens beginning with `@` fail?

### DISCUSSION THREAD

**Discussion by nkcmr** [alecthomas/participle#413] [General]
I am trying to use this library to try to create a Go parser of [Turtle](https://www.w3.org/TR/turtle/). One of the productions of Turtle is a line that literally begins with `@prefix` ([here](https://www.w3.org/TR/turtle/#grammar-production-prefixID)).

When I try to create a parser that specifies that the `"@prefix"` token starts the line, it just doesn't work in Participle. I've condensed this issue into a small, runnable snippet here: https://go.dev/play/p/kOQ2_bAA2Re

This snippet builds 2 parsers:
```
GRAMMAR (broken):
ProgramBroken = WordBroken* .
WordBroken = "@word" <string> ";" .

GRAMMAR (ok):
ProgramOK = WordOK* .
WordOK = "word" <string> ";" .
```

The one that makes use of the `@` symbol does not seem to work, but when the grammar does not use the `@` symbol, it works fine.

I apologize if I have missed something obvious in the documentation, but I did scan around for a few hours to try to get past this issue to no avail. Please let me know if any other info helps, and obviously: thank you so much for this library and your time.

**Comment by alecthomas** [comment]
`@` and `word` are two separate tokens, so you'll need to match them separately like `"@" "word"`

**Comment by nkcmr** [comment]
That does seem to get some things to work. But then it allows for things like `@   word "bar"`, or `@\n\n  word "foo"`; which, applying back to my use-case of trying to implement a Turtle parser, is invalid.

**Comment by alecthomas** [comment]
If you need it to be a single token containing `@<word>`, the solution is to define a custom lexer that captures the entire token.
